src/main.rs

```diff
@@ -239,8 +243,14 @@ fn main() {
 
     let args: Args = Args::docopt().decode().unwrap_or_else(|e| e.exit());
 
-    let local_name = update_hostname(args.flag_local_name.to_owned());
-    let local_name = format!("{}.local", local_name);
+    let local_name = args.flag_local_name;
+    let local_name = update_hostname(&local_name)
+        .or_else(|err| {
+            error!("Could not update local host name: {}", err);
+            Ok::<String, HostManagerError>(local_name)
```

Comment:
Type annotation is here because of compilation error ```unable to infer enough type information about `_`; type annotations or generic parameter binding required [E0282]```